api/vm/compile.ts

```diff
@@ -69,25 +68,20 @@ export function compile(compileConfig: dataform.ICompileConfig) {
 
 export function listenForCompileRequest() {
   process.on("message", (compileConfig: dataform.ICompileConfig) => {
-    const handleError = (e: any) => {
-      const serializableError = {};
-      for (const prop of Object.getOwnPropertyNames(e)) {
-        (serializableError as any)[prop] = e[prop];
-      }
-      process.send(serializableError);
-    }
+    // Initializing writer here results in a `uv_pipe_open` error.
+    let writer: net.Socket;
     try {
       const compiledResult = compile(compileConfig);
-      const writer = new net.Socket({ fd: 4 });
-      writer.write(compiledResult, (err) => {
+      writer = new net.Socket({ fd: 4 });
+      writer.write(compiledResult, (err: any) => {
         if (err) {
-          handleError(err);
+          writer?.end(`Error compiling in child process: ${err?.message}`);
```

Comment:
can you clarify what exactly the conflict is, how it arises, etc?

this fix, i.e. multiplexing the response in a very unstructured way, feels like a bad idea / something that is going to bite us. it also loses any `Error` fields.

Reply:
Having `process.send` present in the code (it doesn't have to be called) after or before socket initialisation causes the `uv_pipe_open` conflict error. Happy for alternatives, but can't find a decent way to send the entire error as is; `writer.destroy` won't send it.